File: algorithm/sort/insertion_sort.c
#include <sys/types.h>

void insertion_sort(int arr[], size_t len)
{
	int i, j;
	int sort_target;

	for (i = 1; i < len; i++)
	{
		sort_target = arr[i];
		for (j = i - 1; j >= 0; j--)
		{
			if (arr[j] > sort_target)
			{
				arr[j + 1] = arr[j];
			}
			else
				break;
		}
		arr[j + 1] = sort_target;
	}
}

File: algorithm/sort/insertion_sort/InsertionSort.java
public class InsertionSort {
	public void insertionSort(int[] arr, int len) {
		int i;
		int j;
		int insertionData;

		for (i = 1; i < len; i++) {
			insertionData = arr[i];
			for (j = i - 1; j >= 0; j--) {
				if (arr[j] > insertionData) {
					arr[j + 1] = arr[j];
				} else {
					break;
				}
			}

			arr[j + 1] = insertionData;
		}
	}
}